hid-xpadneo/src/hid-xpadneo.c

```diff
@@ -899,7 +906,12 @@ static int xpadneo_raw_event(struct hid_device *hdev, struct hid_report *report,
 		bits |= (data[14] & (BIT(0) | BIT(1))) >> 0;	/* A, B */
 		bits |= (data[14] & (BIT(3) | BIT(4))) >> 1;	/* X, Y */
 		bits |= (data[14] & (BIT(6) | BIT(7))) >> 2;	/* LB, RB */
-		bits |= (data[16] & BIT(0)) << 6;	/* Back */
+		if (xdata->quirks & XPADNEO_QUIRK_SHARE_BUTTON) {
+			/* correct button mapping of Xbox controllers with a share button */
+			bits |= (data[15] & BIT(2)) ? BIT(6) : 0; /* Back */
+		} else {
+			bits |= (data[16] & BIT(0)) << 6; /* Back */
+		}
 		bits |= (data[15] & BIT(3)) << 4;	/* Menu */
 		bits |= (data[15] & BIT(5)) << 3;	/* LS */
 		bits |= (data[15] & BIT(6)) << 3;	/* RS */
```

Comment:
The order of the above lines reflects the positional orders of buttons games expect. So the share button should probably somewhere after here, even if that means we need another `if quirk-sharebutton`.

Reply:
Or it doesn't quite so... Menu should move down. Let's ignore that for now and do better from now on. ;-)